Xamarin.Forms.Core/CarouselView/CarouselView.cs

```diff
@@ -0,0 +1,141 @@
+			set { SetValue(IndicatorsShapeProperty, value); }
+		}
+
+		public static readonly BindableProperty PageIndicatorTintColorProperty = BindableProperty.Create("PageIndicatorTintColor", typeof(Color), typeof(CarouselView), Color.Silver);
+
+		public Color PageIndicatorTintColor
+		{
+			get { return (Color)GetValue(PageIndicatorTintColorProperty); }
+			set { SetValue(PageIndicatorTintColorProperty, value); }
+		}
+
+		public static readonly BindableProperty CurrentPageIndicatorTintColorProperty = BindableProperty.Create("CurrentPageIndicatorTintColor", typeof(Color), typeof(CarouselView), Color.Gray);
+
+		public Color CurrentPageIndicatorTintColor
+		{
+			get { return (Color)GetValue(CurrentPageIndicatorTintColorProperty); }
+			set { SetValue(CurrentPageIndicatorTintColorProperty, value); }
+		}
+
+		public static readonly BindableProperty OrientationProperty = BindableProperty.Create("Orientation", typeof(CarouselViewOrientation), typeof(CarouselView), CarouselViewOrientation.Horizontal);
+
+		public CarouselViewOrientation Orientation
+		{
+			get { return (CarouselViewOrientation)GetValue(OrientationProperty); }
+			set { SetValue(OrientationProperty, value); }
+		}
+
+		public static readonly BindableProperty AnimateTransitionProperty = BindableProperty.Create("AnimateTransition", typeof(bool), typeof(CarouselView), true);
+
+		public bool AnimateTransition
+		{
+			get { return (bool)GetValue(AnimateTransitionProperty); }
+			set { SetValue(AnimateTransitionProperty, value); }
+		}
+
+		public static readonly BindableProperty IsSwipingEnabledProperty = BindableProperty.Create("IsSwipingEnabled", typeof(bool), typeof(CarouselView), true);
+
+		public bool IsSwipingEnabled
+		{
+			get { return (bool)GetValue(IsSwipingEnabledProperty); }
+			set { SetValue(IsSwipingEnabledProperty, value); }
+		}
+
+		// Android and iOS only
+		public static readonly BindableProperty InterPageSpacingProperty = BindableProperty.Create("InterPageSpacing", typeof(int), typeof(CarouselView), 0);
+
+		public int InterPageSpacing
+		{
+			get { return (int)GetValue(InterPageSpacingProperty); }
+			set { SetValue(InterPageSpacingProperty, value); }
+		}
+
+		// Android and iOS only
+		public static readonly BindableProperty InterPageSpacingColorProperty = BindableProperty.Create("InterPageSpacingColor", typeof(Color), typeof(CarouselView), Color.White);
+
+		public Color InterPageSpacingColor
+		{
+			get { return (Color)GetValue(InterPageSpacingColorProperty); }
+			set { SetValue(InterPageSpacingColorProperty, value); }
+		}
+
+		// UWP only
+		public static readonly BindableProperty ArrowsProperty = BindableProperty.Create("Arrows", typeof(bool), typeof(CarouselView), false);
+
+		public bool Arrows
+		{
+			get { return (bool)GetValue(ArrowsProperty); }
+			set { SetValue(ArrowsProperty, value); }
+		}
+
+		public EventHandler PositionSelected;
+
+		public Func<object, int, Task> InsertAction;
```

Comment:
Members that are only public because they need to be accessed by the renderer are typically put into ICarouselViewController interface and then implemented explicitly on the element so as to hide them from isense for app developers. We are considering a simpler solution whereby you need only add `[EditorBrowsable(EditorBrowsableState.Never)]` but that PR is limbo at the moment...